src/org/opendatakit/briefcase/ui/reused/AggregateServerConnectionDialog.java

```diff
@@ -0,0 +1,80 @@
+package org.opendatakit.briefcase.ui.reused;
+
+import java.util.function.Consumer;
+import java.util.function.Predicate;
+
+import org.opendatakit.briefcase.model.ServerConnectionInfo;
+import org.opendatakit.briefcase.model.TerminationFuture;
+import org.opendatakit.briefcase.ui.MessageStrings;
+import org.opendatakit.briefcase.ui.ODKOptionPane;
+import org.opendatakit.briefcase.util.ServerConnectionTest;
+
+public class AggregateServerConnectionDialog {
+  final AggregateServerConnectionDialogForm form;
+  private boolean isSuccessful;
+  private ServerConnectionInfo serverInfo = null;
+  private TerminationFuture terminationFuture = new TerminationFuture();
+  private AggregateServerConnectionConfiguration configuration;
+  Predicate<AggregateServerConnectionConfiguration> confValidator;
+
+  AggregateServerConnectionDialog(AggregateServerConnectionDialogForm form, Predicate<AggregateServerConnectionConfiguration> confValidator, AggregateServerConnectionConfiguration configuration) {
```

Comment:
I've prepared an example of what I meant by injecting the predicate here. It's on https://github.com/ggalmazor/briefcase/commit/9084ca5b9e3192d31e5cfbd4e43f42ea084740e1